climada/entity/exposures/crop_production.py

```diff
@@ -481,57 +511,65 @@ def init_full_exposure_set(input_dir=INPUT_DIR, filename=None, hist_mean_dir=HIS
     # create exposures for all crop-irrigation combinations and save them
     filename_list = list()
     output_list = list()
-    for i, _ in enumerate(filenames):
-        item = filenames[i].split('_')
-        cropyield = CropyieldIsimip()
-        cropyield.set_from_single_run(input_dir=input_dir, filename=filename,
-                                      hist_mean=hist_mean_dir, bbox=bbox,
-                                      yearrange=yearrange, crop=((item[2]).split('-'))[0],
-                                      irr=((item[2]).split('-'))[1], unit=unit)
-        filename_saveto = ('cropyield_isimip_' + ((item[2]).split('-'))[0] + '-'
-                           + (((item[2]).split('-'))[1]).split('.')[0] + '_'
-                           + str(yearrange[0]) + '-' + str(yearrange[1]) + '.hdf5')
-        filename_list.append(filename_saveto)
-        output_list.append(cropyield)
-        cropyield.write_hdf5(os.path.join(output_dir, 'Exposure', filename_saveto))
-
-    if returns == 'filename_list':
+    for file in filenames:
+        item = file.split('_')
```

Comment:
`item` is not very clear as name here.

Reply:
Do you have an idea for a better word? I just came up with element or component, but they seem so long..

Reply:
Ideally, you should use a variable name that makes it clear what it is... In this case, what are the elements in file.split('_')? It looks like maybe they are crop names? Or irrigation types? So, something like crop_prop (for crop properties) maybe? Also, try to avoid the hard-coded indexes... it is completely unclear for someone reading the code what is contained in the variable item[0], item[1] and item[2].

Reply:
Yes, it's the crop and irrigation type and the time period of the historical mean, that is used - so maybe histmean_prop? 

I'll try to think of an alternative to match the ISIMIP exposure files to the corresponding historical mean without using hard-coded indexes.. we used them a lot, also in the relative_cropyield hazard.

Reply:
maybe if we leave it hard coded but change it to be a dictionary, e.g. item['crop_model'], item['irrigation_type'], etc? that should be less confusing

Reply:
This is probably suboptimal, as dictionaries are mutable.